tensorflow_io/text/python/ops/text_ops.py

```diff
@@ -77,3 +81,53 @@ def __init__(self, filenames):
 
   def setitem(self, index, item):
     text_ops.text_output_sequence_set_item(self._resource, index, item)
+
+
+def _infer_dtype(val):
+  """_infer_dtype"""
+  try:
+    val = ast.literal_eval(val)
+  except (SyntaxError, ValueError):
+    return tf.string
+  if isinstance(val, int):
+    if np.int32(val) == val:
+      return tf.int32
+    elif np.int64(val) == val:
+      return tf.int64
+  elif isinstance(val, float):
+    if np.float32(val) == val:
+      return tf.float32
+    elif np.float64(val) == val:
+      return tf.float64
+  return tf.string
+
+
+def from_csv(filename, header=0):
+  """Read csv to Dataset
+
+  NOTE: Experimental and eager only!
+
+  Args:
+    filename: A `tf.string` tensor containing filename.
+  """
+  if not tf.executing_eagerly():
+    raise NotImplementedError("from_csv only support eager mode")
+  dataset = TextDataset(filename)
+  columns = None
+  if header is not None:
+    if header != 0:
```

Comment:
combine these two if conditions?

Reply:
@terrytangyuan The two conditions are not exactly combined. As:
- header=None: no column specified, so using `range(number of columns)` as column names
- header=0: Use the first line as header
- header=1+: tf.data.Dataset does not exactly help in this case, so a ValueError has been raised.

Reply:
@terrytangyuan this follows pandas's `read_csv`'s convention.